### ISSUE TITLE
Bug: [rule name here] <short description of the issue>

### ISSUE THREAD

**Issue by markbrouch** [typescript-eslint/typescript-eslint#9421]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.5.2&fileType=.tsx&code=MYewdgzgLgBAtgTwAoCcRwJYQKYHkBGAVtsLALwyrpbYB0A7hlABYBK2EIANgG7YoQAPDxAYAJgD4AFAEoA3EA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1uYDcBDRgBNaPDpWFFS6KImjQO0SODABfECqA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
const myPromiseObject = Promise.withResolvers<void>();
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/no-invalid-void-type": "error"
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

I expected typing `withResolvers()` as `void` to be valid, as it is for `new Promise<void>()`.

### Actual Result

`void is only valid as a return type or generic type argument.` eslint([@typescript-eslint/no-invalid-void-type](https://typescript-eslint.io/rules/no-invalid-void-type))

### Additional Info

This is the suggested way to type a `Promise.withResolvers()` when you want to resolve with 0 arguments: https://github.com/microsoft/TypeScript/pull/56593#discussion_r1466702656

**Comment by Josh-Cena** [comment]
Duplicate of https://github.com/typescript-eslint/typescript-eslint/issues/3612, which in turn is tracked in https://github.com/typescript-eslint/typescript-eslint/issues/8113. Next time please fill out the issue title. Thanks!
